### PR TITLE
Adds error code output to openrtb auction debugs.

### PR DESCRIPTION
**Pull request by hhhjort**
This expands the reporting of errors in debug requests to include error codes. This will allow for some automation in analyzing debug reports, as scripts can focus on the error codes rather than trying to pattern match against error messages.

### CODE REVIEW

File: exchange/exchange.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func errorsToMetric(errs []error) map[pbsmetrics.AdapterError]struct{} {
 	ret := make(map[pbsmetrics.AdapterError]struct{}, len(errs))
 	var s struct{}
 	for _, err := range errs {
-		if err == context.DeadlineExceeded {
+		if _, ok := err.(*errortypes.Timeout); ok {
```
Comment (dbemiller) [inline exchange/exchange.go]: This should be using the `DecodeError` method now.

Comment (hhhjort) [reply exchange/exchange.go]: What do you mean?

**Comment by dbemiller** [comment]
Probably should delete the existing docs, with this new section & update: https://github.com/prebid/prebid-server/blob/master/docs/endpoints/openrtb2/auction.md#bidder-response-times

File: docs/endpoints/openrtb2/auction.md [hunk 2]
```diff
@@ -232,13 +232,45 @@ It will become impossible to fetch bids from Appnexus within that Request.
...
+```
+
+The codes currently defined are:
+
+```
+0	NoErrorCode
+1	TimeoutCode
+2	BadInputCode
+3	BadServerResponseCode
+999 UnknownErrorCode
```
Comment (dbemiller) [inline docs/endpoints/openrtb2/auction.md]: This spacing looks a little weird... is it mixing tabs & spaces?

Comment (hhhjort) [reply docs/endpoints/openrtb2/auction.md]: Hmmm, could be ... did a little cutting and pasting to put that together, it is possbile. It looked ok in my editor, but tabs can certainly vary between editors.

Comment (dbemiller) [reply docs/endpoints/openrtb2/auction.md]: Markdowns are tricky... I tend to just look them up on my branch in github: https://github.com/prebid/prebid-server/blob/error-codes/docs/endpoints/openrtb2/auction.md#bidder-errors

There are visual studio code plugins which you can use to see the rendered version too though.

**Comment by dbemiller** [comment]
This fixes #473

**Pull request merged**